### DISCUSSION TITLE
Getting and saving original images from the user

### DISCUSSION THREAD

**Discussion by ss-pt** [westacks/telebot#65] [Q&A] [answered]
Hello.

Could you give a code example where the bot takes an image file from the user and saves it?

I found something in the tests: https://github.com/westacks/telebot/blob/master/tests/Feature/SendFilesTest.php

```php
public function testGetFile()
{
    $photos = $this->bot->getUserProfilePhotos([
        'user_id' => getenv('TELEGRAM_USER_ID'),
    ]);
    $this->assertInstanceOf(UserProfilePhotos::class, $photos);

    $file = $this->bot->getFile([
        'file_id' => $photos->photos[0][0]->file_id,
    ]);
    $this->assertInstanceOf(File::class, $file);
    $this->assertNotFalse(filter_var($file->url(getenv('TELEGRAM_BOT_TOKEN')), FILTER_VALIDATE_URL));
}
```

Since the user can submit multiple images at once, I wrote the following code:

```php
if (isset($this->update->message->photo) && is_array($this->update->message->photo) && count($this->update->message->photo)) {
    foreach ($this->update->message->photo as $photo) {
        $file = $this->bot->getFile([
            'file_id' => $photo->file_id,
        ]);
        $url = filter_var($file->url(getenv('TELEGRAM_BOT_TOKEN')), FILTER_VALIDATE_URL);
        if (copy($url, ....)) ...
    }
}
```

**In my code, not the original images will be saved, but their small versions. How can I change the example to get and keep only the original versions of the images?**

**Comment by punyflash** [comment, marked as answer]
Telegram clients compresses images and videos attached to messages before sending by default. Your code is right, and sadly you can't get original uncompressed images if user did not send them without compression
![image](https://user-images.githubusercontent.com/14828695/215831042-88d8756e-6b40-4d49-ac47-1c027b075928.png)
